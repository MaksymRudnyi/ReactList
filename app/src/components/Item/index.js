import React from 'react';
import PropTypes from 'prop-types';

import {
  Container, Logo, Content, Name,
} from './styles';

const Item = ({
  name, logo, city, speciality, id,
}) => (
  <Container>
    <Logo logo={logo} />
    <Content>
      <Name>{name}</Name>
      <p>
        City:
        {' '}
        {city}
      </p>
      <p>
        Speciality:
        {speciality.map(special => (
          <span key={special + id}>
            {special}
            {', '}
          </span>
        ))}
      </p>
    </Content>
  </Container>
);

export default Item;

Item.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  logo: PropTypes.string,
  city: PropTypes.string,
  speciality: PropTypes.arrayOf(PropTypes.string),
};
