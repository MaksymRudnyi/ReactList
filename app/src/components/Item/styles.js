import styled from 'styled-components';

const Container = styled.div`
  border: 1px solid #cecece;
  padding: 15px;
  margin-bottom: 15px;
  display: flex;
  flex-direction: row;
  overflow: hidden;
`;
Container.displayName = 'Container';

const Logo = styled.div`
	width: 120px;
	height: 120px;
	min-width: 120px;
	background: #1e1e1e;
	margin-right: 15px;
`;
Logo.displayName = 'Logo';

const Content = styled.div`
	
`;
Content.displayName = 'Content';

const Name = styled.p`
	font-weight: bold;
	font-size: 18px;
	margin-bottom: 10px;
`;
Name.displayName = 'Name';

export {
	Container,
	Logo,
	Content,
	Name,
};
