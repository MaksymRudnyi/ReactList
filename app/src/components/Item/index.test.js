import React from 'react';
import toJson from 'enzyme-to-json';

import { shallow } from 'enzyme';

import Item from './';

describe('Item', () => {
	let sut;
	let props = {
		id: 1,
		name: 'Company one',
		logo: '',
		city: 'Berlin',
		speciality: ['Excavations', 'Plumbing', 'Electrical'],
	};

	it('should match snapshot', () => {
		sut = shallow(<Item {...props} />);

		expect(toJson(sut)).toMatchSnapshot();
	});
});
