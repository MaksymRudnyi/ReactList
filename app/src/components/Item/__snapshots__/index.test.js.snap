// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item should match snapshot 1`] = `
<Container>
  <Logo />
  <Content>
    <Name>
      Company one
    </Name>
    <p>
      City: 
      Berlin
    </p>
    <p>
      Speciality: 
      <span
        key="Excavations"
      >
        Excavations
        , 
      </span>
      <span
        key="Plumbing"
      >
        Plumbing
        , 
      </span>
      <span
        key="Electrical"
      >
        Electrical
        , 
      </span>
    </p>
  </Content>
</Container>
`;
